Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -290,6 +290,72 @@ For white-box source code testing, you'll need a setup similar to the developer'
+
+With Ghidra, strings can be obtained by simply loading the DEX file and selecting **Window -> Defined strings** in the menu.
+
+> Loading an APK file directly into Ghidra might lead to inconsistencies. Thus it is recommended to extract the DEX file by unzipping the APK file and then loading it into Ghidra.
+
+<img src="Images/Chapters/0x05c/ghidra_dex_strings.png" width="500px"/>
+
+
+With Dextra, you can dump all the strings using the following command:
+
+```bash
+dextra -S classes.dex
+```
+
+The output from Dextra can be manipulated using standard Linux commands, for example, using `grep` to search for certain keywords.
+
+It is important to know, the list of strings obtained using the above tools can be very big, as it also includes the various class and package names used in the application. Going through the complete list, specially for big binaries, can be very cumbersome. Thus, it is recommended to start with keyword-based searching and go through the list only when keyword search does not help. Some generic keywords which can be a good starting point are - password, key, and secret. Other useful keywords specific to the context of the app can be obtained while you are using the app itself. For instance, imagine that the app has as login form, you can take note of the displayed placeholder or title text of the input fields and use that as an entry point for your static analysis.
+
+###### Native Code
+
+In order to extract strings from native code used in an Android application, you can use GUI tools such as Ghidra or Cutter or rely on CLI-based tools such as the _strings_ Unix utility (`strings <path_to_binary>`) or radare2's rabin2 (`rabin2 -zz <path_to_binary>`). When using the CLI-based ones you can take advantage of other tools such as grep (e.g. in conjunction with regular expressions) to further filter and analyze the results. 
+
+##### Cross References
+
+###### Java and Kotlin
+
+Java cross references is supported by many tools, including Ghidra and [jadx](https://github.com/skylot/jadx "jadx"). With Ghidra cross references can be obtained by right clicking on the desired function in the symbol tree and selecting **Show References to**, as shown in the image below.
+
+<img src="Images/Chapters/0x05c/Ghidra_java_xrefs.png" width="500px"/>
+
+<img src="Images/Chapters/0x05c/jadx_java_xrefs.png" width="500px"/>
```

Comment:
As these options are pretty basic I'd remove the pictures, for maintainability: if they ever change the names of the options it'd be easier to update just the text. I've also included a link to a gif from the official jadx Wiki showing this live.
```suggestion
There are many RE tools that support retrieving Java cross references. For many of the GUI-based ones, this is usually done by right clicking on the desired function and selecting the corresponding option, e.g. **Show References to** in Ghidra or [**Find Usage** in jadx](https://github.com/skylot/jadx/wiki/jadx-gui-features-overview#find-usage "jadx").
```